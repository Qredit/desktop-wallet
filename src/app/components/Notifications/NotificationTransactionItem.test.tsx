import { Contracts, DTO } from "@payvo/sdk-profiles";
import userEvent from "@testing-library/user-event";
import nock from "nock";
import React from "react";

import { NotificationTransactionItem } from "./NotificationTransactionItem";
import { httpClient } from "@/app/services";
import { TransactionFixture } from "@/tests/fixtures/transactions";
import { env, getDefaultProfileId, render, screen, waitFor } from "@/utils/testing-library";

const NotificationTransactionsFixtures = require("tests/fixtures/coins/ark/devnet/notification-transactions.json");
const TransactionsFixture = require("tests/fixtures/coins/ark/devnet/transactions.json");

let profile: Contracts.IProfile;
let notificationTransaction: DTO.ExtendedConfirmedTransactionData;

describe("Notifications", () => {
	beforeEach(async () => {
		httpClient.clearCache();

		nock("https://ark-test.payvo.com").get("/api/transactions").query(true).reply(200, {
			data: NotificationTransactionsFixtures.data,
			meta: TransactionsFixture.meta,
		});

		profile = env.profiles().findById(getDefaultProfileId());

		await env.profiles().restore(profile);
		await profile.sync();
		await profile.notifications().transactions().sync();
		notificationTransaction = profile.notifications().transactions().transaction(TransactionFixture.id());
	});

	it("should render notification item", async () => {
		const { container } = render(
			<table>
				<tbody>
					<NotificationTransactionItem transaction={notificationTransaction!} profile={profile} />
				</tbody>
			</table>,
		);
		await waitFor(() => expect(screen.getAllByTestId("TransactionRowMode")).toHaveLength(1));

		expect(container).toMatchSnapshot();
	});

	it("should render notification item with wallet alias", async () => {
		render(
			<table>
				<tbody>
					<NotificationTransactionItem transaction={notificationTransaction!} profile={profile} />
				</tbody>
			</table>,
		);
		await waitFor(() => expect(screen.getAllByTestId("TransactionRowMode")).toHaveLength(1));

		expect(screen.getByTestId("Address__alias")).toHaveTextContent("ARK Wallet 1");
	});

	it("should emit onVisibilityChange event", async () => {
		const onVisibilityChange = jest.fn();

		render(
			<table>
				<tbody>
					<NotificationTransactionItem
						transaction={notificationTransaction!}
						profile={profile}
						onVisibilityChange={onVisibilityChange}
					/>
				</tbody>
			</table>,
		);
		await waitFor(() => expect(screen.getAllByTestId("TransactionRowMode")).toHaveLength(1));
		await waitFor(() => expect(onVisibilityChange).toHaveBeenCalledWith(expect.any(Boolean)));
	});

	it("should emit events onTransactionClick", async () => {
		const onTransactionClick = jest.fn();

		render(
			<table>
				<tbody>
					<NotificationTransactionItem
						transaction={notificationTransaction!}
						profile={profile}
						onTransactionClick={onTransactionClick}
					/>
				</tbody>
			</table>,
		);
		await waitFor(() => expect(screen.getAllByTestId("TransactionRowMode")).toHaveLength(1));

		userEvent.click(screen.getByTestId("TransactionRowMode"));

		await waitFor(() => expect(onTransactionClick).toHaveBeenCalledWith(notificationTransaction));
	});
});
