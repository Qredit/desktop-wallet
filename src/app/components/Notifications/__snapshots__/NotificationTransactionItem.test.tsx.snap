// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications should render notification item 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="TableRow__TableRowStyled-sc-y9j4av-0 ijbCHm group"
        data-testid="TableRow"
      >
        <td
          class="w-3/5"
        >
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 beVmY flex space-x-3"
          >
            <div
              class="flex items-center -space-x-1"
              data-testid="TransactionRowMode"
            >
              <div
                class="Circle__CircleWrapper-sc-d5u41q-0 egSsGQ ring-theme-background group-hover:ring-theme-secondary-100 group-hover:bg-theme-secondary-100 dark:group-hover:ring-black dark:group-hover:bg-black border-theme-success-200 text-theme-success-600 dark:border-theme-success-600 ring-theme-background"
              >
                <div
                  class="sc-bdvvtL qGXhc"
                  height="20"
                  width="20"
                >
                  <svg>
                    received.svg
                  </svg>
                </div>
              </div>
              <div
                class="Avatar__AvatarWrapper-sc-18vw8a4-0 ewMeZp ring-theme-background bg-theme-background group-hover:ring-theme-secondary-100 group-hover:bg-theme-secondary-100 dark:group-hover:ring-black dark:group-hover:bg-black flex-shrink-0"
                data-testid="Avatar"
              >
                <div
                  class="w-full h-full inline-flex items-center justify-center overflow-hidden align-middle rounded-full"
                >
                  <img
                    alt="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                    src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(0, 150, 136)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"50\\" cx=\\"80\\" cy=\\"50\\" fill=\\"rgb(76, 175, 80)\\"/><circle r=\\"35\\" cx=\\"100\\" cy=\\"40\\" fill=\\"rgb(156, 39, 176)\\"/><circle r=\\"40\\" cx=\\"90\\" cy=\\"70\\" fill=\\"rgb(3, 169, 244)\\"/></svg>"
                    title="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                  />
                </div>
              </div>
            </div>
            <div
              class="w-20 flex-1"
            >
              <div
                class="flex overflow-hidden flex-grow items-center space-x-2 whitespace-nowrap no-ligatures"
              >
                <span
                  class="font-semibold text-base text-theme-text"
                  data-testid="Address__alias"
                >
                  <span
                    class="TruncateEnd__Wrapper-sc-1hhdnni-0"
                    data-testid="TruncateEnd"
                  >
                    ARK Wallet 1
                  </span>
                </span>
                <span
                  class="flex-1 inline-flex overflow-hidden text-theme-secondary-500 dark:text-theme-secondary-700 font-semibold text-base"
                  data-testid="Address__address"
                >
                  D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD
                </span>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 jEaNbo justify-end"
          >
            <div
              class="Label-sc-4akkzw-0 hhvRsd"
              color="success"
            >
              <div
                class="flex space-x-1 items-stretch"
              >
                <span
                  class="whitespace-nowrap"
                  data-testid="Amount"
                >
                  + 20 DARK
                </span>
              </div>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
